test/test_rng.cpp

```diff
@@ -0,0 +1,39 @@
+/* Copyright (C) 2021, Nikolai Wuttke. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <base/warnings.hpp>
+#include <engine/random_number_generator.hpp>
+
+RIGEL_DISABLE_WARNINGS
+#include <catch.hpp>
+RIGEL_RESTORE_WARNINGS
+
+TEST_CASE("Random number generator works the same as the original") {
+  rigel::engine::RandomNumberGenerator rng;
+  SECTION("Values are generated in this particular order") {
+    SECTION("First generated value is 8") {
+      CHECK(rng.gen() == 8u);
+    }
+    SECTION("The period of the RNG is the original array's size") {
```

Comment:
I think this is a good choice of what to test, period and start value makes sense. But I'd suggest making this test a bit stricter: If the period would somehow be different, the test would still pass as long as the 257th value is the same as the 1st value.

One way to make it stricter could be to generate `RANDOM_NUMBER_TABLE.size() * 2` values into a `std::vector`, and then check that the first 256 values are identical to the 2nd half of the vector.

What do you think?

Reply:
I think I covered what you described.
Was a fun test to write :)